### PR TITLE
add belongs_to primary_id (aka. primary_key in rails) support

### PR DESCRIPTION
**Pull request by u007**
as described here in rails
https://apidock.com/rails/ActiveRecord/Associations/ClassMethods/belongs_to

### CODE REVIEW

File: associations/belongs_to_association.go [hunk 1]
```diff
@@ -40,17 +50,40 @@ func belongsToAssociationBuilder(p associationParams) (Association, error) {
 	if fieldIsNil(f) || isZero(f.Interface()) {
 		skipped = true
 	}
+	//associated model
+	ownerPrimaryTableField := "id"
+	if primaryIDField != "ID" {
+		var ownerModel reflect.Value
```
Comment (larrymjordan) [inline associations/belongs_to_association.go]: @u007 Just one thing I noticed, you should probably use [reflect.Indirect](https://golang.org/pkg/reflect/#Indirect) method instead. And you would save these conditionals statements.  Beside this I think this looks good.

Comment (u007) [reply associations/belongs_to_association.go]: hi @larrymjordan , did u mean this line?
```
if fval.Kind() == reflect.Ptr {
	ownerModel = fval.Elem()
} else {
	ownerModel = fval
}

```

sorry im not really sure how indirect is used. With Indirect, does it means i dont have to check for pointer?

Comment (larrymjordan) [reply associations/belongs_to_association.go]: @u007 That's correct, in this case,`Indirect` would return the value that `fval` is pointing to, if fval is a pointer.

Comment (u007) [reply associations/belongs_to_association.go]: done :) pls check if its good

**Comment by stanislas-m** [comment]
@u007 I'll take a look soon, but why not using `primary_key`? Is there a reason to use `primary_id` instead? :)

**Comment by u007** [comment]
hi @stanislas-m cox we are already using fk_id

**Pull request merged**